src/components/History/History.tsx

```diff
@@ -1,62 +1,243 @@
-import React, {
-  useEffect,
-  useState,
-} from 'react';
-
-import {
-  getItem,
-} from '../../services/storage';
-import { STORED_GROUPED_CROSS_CHAIN_ACTIONS } from '../../constants/storageConstants';
-import ActionPreview from '../TransactionPreview/ActionPreview';
+import React, { useEffect, useState } from 'react';
+import styled, { useTheme } from 'styled-components';
+
+import HistoryPreview from '../TransactionPreview/HistoryPreview';
 import { ICrossChainAction } from '../../types/crossChainAction';
 
-const History = () => {
-  const [storedGroupedCrossChainActions, setStoredGroupedCrossChainActions] = useState<{ [id: string]: ICrossChainAction[] }>({})
-
-  const getLatest = () => {
-    try {
-      const storedGroupedCrossChainActionsRaw = getItem(STORED_GROUPED_CROSS_CHAIN_ACTIONS);
-      const storedGroupedCrossChainActionsUpdated = storedGroupedCrossChainActionsRaw
-        ? JSON.parse(storedGroupedCrossChainActionsRaw)
-        : {};
-      setStoredGroupedCrossChainActions(storedGroupedCrossChainActionsUpdated);
-    } catch (e) {
-      //
+// Hooks
+import { useEtherspot } from '../../hooks';
+
+//utils
+import { Theme } from '../../utils/theme';
+import ContentLoader from 'react-content-loader';
+import MenuModalWrapper from '../Menu/MenuModalWrapper';
+
+//constants
+import { TRANSACTION_BLOCK_TYPE } from '../../constants/transactionBuilderConstants';
+import { CROSS_CHAIN_ACTION_STATUS } from '../../constants/transactionDispatcherConstants';
+
+import { getNativeAssetPriceInUsd } from '../../services/coingecko';
+
+import { uniqueId } from 'lodash';
+
+const History = ({ onBackButtonClick }: { onBackButtonClick: () => void }) => {
+  const [storedGroupedCrossChainActions, setStoredGroupedCrossChainActions] = useState<{
+    [id: string]: ICrossChainAction[];
+  }>({});
+  const [storedTransactionsDetails, setStoredTransactionsDetails] = useState<{
+    [id: string]: ICrossChainAction[];
+  }>({});
+  const [isLoadingTransactions, setIsLoadingTransactions] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const { getAllTransactions, getSupportedAssetsForChainId, accountAddress } = useEtherspot();
+  const theme: Theme = useTheme();
+
+  const pullAllNewTx = async (accountAddress: string | null) => {
+    setIsLoadingTransactions(true);
+    const transactions = await getAllTransactions(accountAddress);
+    if (!transactions) {
+      setErrorMessage('Failed to get transactions!');
+      return;
     }
+    let crossChainAction: ICrossChainAction[] = [];
+    transactions &&
+      Object.entries(transactions).forEach(async ([chain_id, value]) => {
```

Comment:
I used only those one which is necessary in optimised way so, it will work well

Reply:
So it's less about whether it will work well but more about code hygiene, cyclomatic complexity, readability for other developers and testability. When you say `in optimised way`, how is it optimised?

Reply:
Only setting and manipulating those fields which are needed in TX history